Source/WebCore/layout/formattingContexts/inline/InlineContentBalancer.cpp

```diff
@@ -110,7 +110,8 @@ InlineContentBalancer::InlineContentBalancer(InlineFormattingContext& inlineForm
 
 void InlineContentBalancer::initialize()
 {
-    if (!m_inlineFormattingContext.layoutState().placedFloats().isEmpty()) {
+    auto numberOfVisibleLinesAllowed =  m_inlineFormattingContext.numberOfVisibleLinesAllowed();
+    if (!m_inlineFormattingContext.layoutState().placedFloats().isEmpty() || numberOfVisibleLinesAllowed == 1) {
         m_cannotBalanceContent = true;
```

Comment:
I think this is quite misleading and somewhat error prone. numberOfVisibleLinesAllowed == 1 means we **_should not_** balance which is very different from cannot. -even if InlineContentBalancer currently ends up doing the same for both cases.

Reply:
Agree. I've almost changed the name to `shouldNotBalanceContent`. I guess I'll do it. I assume that "should not" also makes sense for the case we just to avoid balancing for floats until it is unsupported.